compiler/resolution/functionResolution.cpp

```diff
@@ -1047,8 +1047,10 @@ static bool canParamCoerce(Type*   actualType,
     }
   }
 
+  // need to allow converting param string->c_ptrConst(c_char)
   // param strings can coerce between string and c_string
-  if ((formalType == dtString || formalType == dtStringC) &&
+  if ((formalType == dtString || formalType == dtStringC ||
+       isCPtrConstChar(formalType))                      &&
```

Comment:
This allows implicit conversion from param strings to `c_ptrConst(c_char)`. It seems important in the near term but in the long term, it is unclear to me if we want to do that. Has it been discussed directly? Is there an issue about it?

Reply:
In particular, I could imagine deciding to allow this for passing to `extern proc`s only.

Reply:
I have opened https://github.com/chapel-lang/chapel/issues/22910 to discuss and decide on this question